Feat/solution

eliran17e opened the pull request.

eliran17e committed:
that the way

eliran17e committed:
no vinnigrete

eliran17e committed:
piece of cake

eliran17e committed:
cup of join

eliran17e committed:
parsle tounge

eliran17e committed:
communcation vessels

eliran17e committed:
running 2000

eliran17e committed:
long cat is long

eliran17e committed:
remember remember

eliran17e committed:
group by

eliran17e committed:
sent turtle

eliran17e committed:
no vinnigerette

eliran17e committed:
Delete src/5.1.no_vinnigrete.py

eliran17e committed:
Update and rename 5.1.no_vinnigerette.py to 5.1.no_vinnigrette.py

eliran17e committed:
Update and rename 5.1.that_the_way.py to 5.1.thats_the_way.py

eliran17e committed:
Update 5.1.thats_the_way.py

eliran17e committed:
Update 5.2.cup_of_join.py

eliran17e committed:
Update 5.2.piece_of_cake.py

eliran17e committed:
Update 5.3.parsle_tounge.py

eliran17e committed:
Update and rename 5.3.parsle_tounge.py to 5.3.parsle_tongue.py

eliran17e committed:
Update 6.2.running_2000.py

eliran17e committed:
Update 6.3.long_cat_is_long.py

eliran17e committed:
Update 6.4.remember_remember.py

eliran17e committed:
Update 6.4.remember_remember.py

eliran17e committed:
Update 6.4.remember_remember.py

eliran17e committed:
Update 5.1.no_vinnigrette.py

eliran17e committed:
Update 5.1.no_vinnigrette.py

eliran17e committed:
Update 5.1.thats_the_way.py

eliran17e committed:
Update 5.1.thats_the_way.py

eliran17e committed:
Update 5.1.thats_the_way.py

eliran17e committed:
Update 5.2.cup_of_join.py

eliran17e committed:
Update 5.2.cup_of_join.py

eliran17e committed:
Update 5.2.cup_of_join.py

eliran17e committed:
Update 5.2.cup_of_join.py

eliran17e committed:
Update 5.3.parsle_tongue.py

eliran17e committed:
Update 5.3.parsle_tongue.py

eliran17e committed:
logo.jpg

eliran17e committed:
code.png

eliran17e committed:
Update 5.3.parsle_tongue.py

eliran17e committed:
Update 5.3.parsle_tongue.py

eliran17e committed:
Update 5.4.communicating_vessels.py

eliran17e committed:
Update 5.4.communicating_vessels.py

eliran17e committed:
Update 5.4.communicating_vessels.py

eliran17e committed:
Update 6.2.running_2000.py

eliran17e committed:
Update 6.2.running_2000.py

eliran17e committed:
Update 7.2.sent_turtle.py

eliran17e committed:
Update 7.2.sent_turtle.py

eliran17e committed:
Update 7.2.sent_turtle.py

eliran17e committed:
Update 5.1.no_vinnigrette.py

eliran17e committed:
Update 5.1.no_vinnigrette.py

eliran17e committed:
Update 5.1.no_vinnigrette.py

eliran17e committed:
Update 5.1.no_vinnigrette.py

eliran17e committed:
Update 5.1.no_vinnigrette.py

eliran17e committed:
Update 5.1.thats_the_way.py

eliran17e committed:
Update 5.2.cup_of_join.py

eliran17e committed:
Update 5.2.piece_of_cake.py

eliran17e committed:
Update 5.3.parsle_tongue.py

eliran17e committed:
Update 5.4.communicating_vessels.py

eliran17e committed:
Update 6.2.running_2000.py

eliran17e committed:
Update 6.3.long_cat_is_long.py

eliran17e committed:
Update 6.4.remember_remember.py

eliran17e committed:
Update 6.5.group_by.py

eliran17e committed:
Update 7.2.sent_turtle.py

eliran17e committed:
Update 7.2.sent_turtle.py

eliran17e committed:
Update 7.2.sent_turtle.py

eliran17e committed:
Update 7.2.sent_turtle.py

eliran17e committed:
Update 7.2.sent_turtle.py

eliran17e committed:
Update 7.2.sent_turtle.py

excellenteam-scaleup commented on src/5.1.no_vinnigrette.py:
```diff
@@ -0,0 +1,49 @@
+    between them is a Monday.
+    """
+    dates = []
+    while True:
```
This is  unnecessary use of a loop. Additionally, if you are using a while loop, we do not recommend using while TRUE,A while loop should ideally have a clear exit condition
-1

excellenteam-scaleup commented on src/5.1.no_vinnigrette.py:
```diff
@@ -0,0 +1,49 @@
+    date2 (datetime or date): The other boundary date
+    Returns true if its monday, false otherwise
+    """
+    if isinstance(date1, str):
```
Is this check really necessary? you already chaked the user input in Line 40

excellenteam-scaleup commented on src/5.1.thats_the_way.py:
```diff
@@ -0,0 +1,38 @@
+    """
+    relative_path = "./images"
+    full_path = os.path.abspath(relative_path)
+    print("Full path to the directory:", full_path)
```
Unnecessary use of print

excellenteam-scaleup commented on src/5.1.no_vinnigrette.py:
```diff
@@ -0,0 +1,49 @@
+
+
+
+def no_vinnigrete(date1, date2):
```
It is customary to describe the types of variables that the function will accept

excellenteam-scaleup commented on src/5.1.no_vinnigrette.py:
```diff
@@ -0,0 +1,49 @@
+    no_vinnigrete(dates[0], dates[1])
+
+if __name__ == "__main__":
+    main()
```
7/9

excellenteam-scaleup commented on src/5.1.no_vinnigrette.py:
```diff
@@ -0,0 +1,49 @@
+    delta = (end_date - start_date).days
+    random_days = random.randint(0, delta)
+    date_result = start_date + datetime.timedelta(days=random_days)
+    if date_result.weekday() == 0:
```
Do not use "Magic numbers", best practice is to assign to a CONSTANT
-1

excellenteam-scaleup commented on src/5.1.thats_the_way.py:
```diff
@@ -0,0 +1,38 @@
+    print(thats_the_way(full_path))
+
+if __name__ == '__main__':
+    main()
```
9/9

excellenteam-scaleup commented on src/5.2.cup_of_join.py:
```diff
@@ -0,0 +1,36 @@
+This module provides a function to combine multiple lists into one list with a separator between them.
+"""
+
+def cup_of_join(*lists, sep=None):
```
If sep is not provided should concatenate "-" between collections...
-4

excellenteam-scaleup commented on src/5.2.cup_of_join.py:
```diff
@@ -0,0 +1,36 @@
+    print(cup_of_join(list1, list2, list3, sep='@'))
+
+if __name__ == '__main__':
+    main()
```
5/9

excellenteam-scaleup replied:
9/9
the mistake in the tests is ours, the grade will be corrected accordingly

excellenteam-scaleup commented on src/5.2.piece_of_cake.py:
```diff
@@ -0,0 +1,39 @@
+    print(piece_of_cake({}))  # Output: 0
+
+if __name__ == '__main__':
+    main()
```
9/9

excellenteam-scaleup commented on src/5.3.parsle_tongue.py:
```diff
@@ -0,0 +1,68 @@
+    base_dir = os.path.dirname(os.path.abspath(__file__))
+    file_path = os.path.join(base_dir, "logo.jpg")
+
+    try:
```
Redundant try/catch expression... please read:
https://builtin.com/software-engineering-perspectives/what-is-with-statement-python
-1

excellenteam-scaleup commented on src/5.3.parsle_tongue.py:
```diff
@@ -0,0 +1,68 @@
+        print(message)
+
+if __name__ == '__main__':
+    main()
```
8/9

excellenteam-scaleup commented on src/5.4.communicating_vessels.py:
```diff
@@ -0,0 +1,50 @@
+"""
+
+def interleave(*lists):
+    """
```
Use list comprehension, leveraging zip_largest
-1

excellenteam-scaleup commented on src/5.4.communicating_vessels.py:
```diff
@@ -0,0 +1,50 @@
+
+def generator_interleave(*lists):
+    """
+    Interleave multiple lists into one list using a generator.
```
Use list comprehension, leveraging zip_largest
-1

excellenteam-scaleup commented on src/5.4.communicating_vessels.py:
```diff
@@ -0,0 +1,50 @@
+    print(list(generator_interleave(list1, list2, list3)))
+
+if __name__ == '__main__':
+    main()
```
7/9

excellenteam-scaleup commented on src/6.2.running_2000.py:
```diff
@@ -0,0 +1,56 @@
+    an error message and returns None instead of crashing.
+    """
+    try:
+        start = time.time()
```
Why did you decide that it should return the result in seconds? Is it accurate enough for fast functions?
-1

excellenteam-scaleup commented on src/6.2.running_2000.py:
```diff
@@ -0,0 +1,56 @@
+        result = end - start
+        return result
+    except (TypeError, ValueError, RuntimeError) as e:
+        print(f"An error occurred while running the function: {e}")
```
raise and not print here
-1

excellenteam-scaleup commented on src/6.2.running_2000.py:
```diff
@@ -0,0 +1,56 @@
+        print("The function failed to run.")
+
+if __name__ == '__main__':
+    main()
```
7/9

excellenteam-scaleup commented on src/6.3.long_cat_is_long.py:
```diff
@@ -0,0 +1,35 @@
+    print(word_count)
+
+if __name__ == '__main__':
+    main()
```
9/9

excellenteam-scaleup commented on src/6.4.remember_remember.py:
```diff
@@ -0,0 +1,57 @@
+    width, height = img.size
+
+    message = ""
+    for x in range(width):
```
Use list comprehension
-1

excellenteam-scaleup commented on src/6.4.remember_remember.py:
```diff
@@ -0,0 +1,57 @@
+    """
+    relative_path = "./code.png"
+    full_path = os.path.abspath(relative_path)
+    if not os.path.exists(full_path):
```
Missing error handling,
what about try/except?
-1

excellenteam-scaleup commented on src/6.4.remember_remember.py:
```diff
@@ -0,0 +1,57 @@
+    print("Decoded message:", message)
+
+if __name__ == '__main__':
+    main()
```
7/9

excellenteam-scaleup commented on src/6.5.group_by.py:
```diff
@@ -0,0 +1,26 @@
+    print(group_by(len, ["hi", "bye", "yo", "try"]))
+
+if __name__ == '__main__':
+    main()
```
Great :)
9/9

excellenteam-scaleup commented on src/7.2.sent_turtle.py:
```diff
@@ -0,0 +1,85 @@
+            raise KeyError(f"User '{recipient}' does not exist.")
+
+        user_box = self.boxes[recipient]
+        self.message_id += 1
```
Better update counter after actual insertion

excellenteam-scaleup commented on src/7.2.sent_turtle.py:
```diff
@@ -0,0 +1,85 @@
+        for message in user_box:
+            if search_string in message['title'].lower() or search_string in message['body'].lower():
+                result.append(message)
+        return result
```
9/9

eliran17e committed:
fixed

removed unnecessery loop and used MONDAY as a const = 0 for more clear code

eliran17e committed:
fixed

made a comprehension list instead of the 2 for loops

eliran17e committed:
fixed

did a try and catch in main function instead of in running. time taken now in milliseconds instead of seconds

eliran17e committed:
fixed

used list comprehension and added try and catch in main

eliran17e committed:
Update 5.4.communicating_vessels.py

excellenteam-scaleup reviewed (commented):
+5 bug fixed
